lib/rspec/core/reporter.rb

```diff
@@ -79,6 +88,17 @@ def message(message)
       notify :message, Notifications::MessageNotification.new(message)
     end
 
+    # @param event [Symbol] Name of the custom event to trigger on formatters
+    # @param options [Hash] Hash of arguments to provide via `CustomNotification`
+    #
+    # Publish a custom event to supporting registered formatters.
+    # @see RSpec::Core::Notifications::CustomNotification
+    def publish(event, options={})
+      event_name = event.to_sym
+      return false if NOTIFICATIONS.include? event_name
```

Comment:
IMO, we should raise an error rather than returning `false`.  In general, `publish` doesn't have meaningful return value and it would be too easy for a user to miss the return value and assume it had worked when it didn't.  It would be particularly frustrating for a user that used this to publish a custom event that in a later release became a lifecycle event...their publication of their custom event would simply start to fail silently.